Sources/SwiftParser/Expressions.swift

```diff
@@ -512,13 +512,19 @@ extension Parser {
   }
 
   @_spi(RawSyntax)
-  public mutating func parseDottedExpressionSuffix() -> (
-      period: RawTokenSyntax, name: RawTokenSyntax,
-      declNameArgs: RawDeclNameArgumentsSyntax?,
-      generics: RawGenericArgumentClauseSyntax?
+  public mutating func parseDottedExpressionSuffix<R: RawSyntaxNodeProtocol>(leading: R?) -> (
+    unexpectedPeriod: RawUnexpectedNodesSyntax?,
+    period: RawTokenSyntax,
+    name: RawTokenSyntax,
+    declNameArgs: RawDeclNameArgumentsSyntax?,
+    generics: RawGenericArgumentClauseSyntax?
   ) {
-    assert(self.at(any: [.period, .prefixPeriod]))
-    let period = self.consumeAnyToken(remapping: .period)
+    assert(self.at(.period))
+    let (unexpectedPeriod, period) = self.consumeMemberPeriod(leading: leading)
+    if currentToken.isAtStartOfLine {
+      let missingIdentifier = RawTokenSyntax(missing: .identifier, arena: self.arena)
```

Comment:
If you want to, you can shorten this to something like

```suggestion
      let missingIdentifier = missingToken(.identifier)
```

Reply:
Ah, sure, makes sense I guess. Though `RawTokenSyntax(missing: .identifier, arena: self.arena)` isn't that bad to me 😅